src/UrlSearchParams.ts

```diff
@@ -0,0 +1,151 @@
+import { duplicate } from './lang';
+
+/**
+ * A object with string keys and string or string array values that describes a query string.
+ */
+type ParamList = { [key: string]: string | string[] };
+
+/**
+ * Parse a query string, returning a ParamList object.
+ */
+function parseQueryString(input: string): ParamList {
+	const query = <{ [key: string]: string[] }> {};
+	input.split('&').forEach(entry => {
```

Comment:
`for...of` seems like a good fit. I feel like our style guide is giving up one of the main benefits of arrow functions, though, in not recommending them for their compactness (which is kind of the whole point of the syntax).